### ISSUE TITLE
[REQUEST] USB CDC detect line_coding & Arduino like reboot detection

### ISSUE THREAD

**Issue by vanbwodonk** [tinyvision-ai-inc/pico-ice#40]
I modified pico_ice_default and pico-ice-sdk/examples/pico_usb_uart with line_coding detection and 1200 bitrate detection for autoreboot. This is very useful if we need a higher bitrate than 115200, and Arduino like auto-reboots when we need to reflash the RP2040. Just need to add some code like this:
```
void tud_cdc_line_coding_cb(uint8_t itf, cdc_line_coding_t const *coding) {
  if (itf == 0) {
    if (coding->bit_rate != baud_rate) {
      baud_rate = coding->bit_rate;
      uart_set_baudrate(uart0, baud_rate);
    }
  }
}

void tud_cdc_line_state_cb(uint8_t itf, bool dtr, bool rts) {
  if (itf == 0) {
    if (rts) {
    }
    if (dtr) {
    } else {
        cdc_line_coding_t coding;
        tud_cdc_get_line_coding(&coding);
        if (coding.bit_rate == 1200) {
          reset_usb_boot(0, 0);
          while (1); 
        }
    }
  }
}
```

Or, if you allow I can make a PR about it.

**Comment by josuah** [comment]
Thank you for this! The way it is implemented makes sense to me, and easy to make that configurable with just an `#ifdef` and [a config in `tusb_config.h`](https://pico-ice.tinyvision.ai/group__ice__usb.html).

**Comment by josuah** [comment]
I am now back from Embedded World and will be able to switch over to integrating this, likely tomorrow.
Thanks everyone for your patience!

**Comment by josuah** [comment]
Adjusting the baud rate works well with what you provided.
I edited it a tiny bit for matching the autoconfig.

Before, baud 1200
![2024-04-20-172334_1549x191_scrot](https://github.com/tinyvision-ai-inc/pico-ice/assets/55351021/1160e702-d827-428c-85ec-5a3b38a6c867)

After, baud 4800 (mismatching Pulseview configuration)
![2024-04-20-172348_1513x204_scrot](https://github.com/tinyvision-ai-inc/pico-ice/assets/55351021/c5b8bd32-4adf-4bc9-a5cd-6aa9dd754675)

**Comment by MrJake222** [comment]
Nice! Does it support resetting and programming with `picotool` (`-f` option)?

**Comment by josuah** [comment]
> Does it support resetting and programming with `picotool` (`-f` option)?

Not yet, but https://arduino.github.io/arduino-cli/dev/platform-specification/#1200-bps-bootloader-reset will be supported

**Comment by josuah** [comment]
I confirm that after using `picocom -b 1200 /dev/ttyACM0` it is possible to use `picotool` though:

```
$ picotool info
Program Information
 name:      pico_ice_default
 features:  USB stdin / stdout
```

**Comment by josuah** [comment]
This landed in v1.6.1, closing :)

**Comment by MrJake222** [comment]
Great! I think you only forgot to create new github release with binaries to download

**Comment by josuah** [comment]
It was in preparation at that moment, and landed here: https://github.com/tinyvision-ai-inc/pico-ice/releases

**Comment by MrJake222** [comment]
> This landed in v1.6.1, closing :)

Did it? Can confirm it works in downloaded `.uf2` but I cannot find any commits or code to build it myself.... Am i blind or you didn't push something?

**Comment by josuah** [comment]
Good catch! 

As seen here, `main` was not pushed to `origin/main`, only `develop` to `origin/develop`
```
* 8d0fd06 (HEAD -> develop, tag: v1.6.1, origin/develop, main) implement tud_cdc_line_coding_cb
* 8e92021 (tag: v1.6.0, origin/main, origin/HEAD) stop ICE40 FPGA before initializing SPI flash
```

I pushed it now: https://github.com/tinyvision-ai-inc/pico-ice-sdk/blob/main/src/ice_usb.c#L317
